app/models/paypal_record.rb

```diff
@@ -50,6 +61,11 @@ class PaypalRecord < ApplicationRecord
   belongs_to :parent_record, class_name: "PaypalRecord", inverse_of: :child_records, optional: true
   has_many :child_records, class_name: "PaypalRecord", inverse_of: :parent_record, foreign_key: :parent_record_id
 
+  def determine_wca_status
+    result = WCA_TO_PAYPAL_STATUS_MAP.find { |key, values| values.include?(self.paypal_status) }
+    result&.first || raise("No associated wca_status for paypal_status: #{self.paypal_status} - our tests should prevent this from happening!")
```

Comment:
Do we have a test that prevents this from happening? I believe we have a test for Stripe that asserts that all wca_statuses have a corresponding stripe_status, but I don't see once in this PR for paypal (apologies if I'm just being dense)

Reply:
Tests added.